src/linear_algebra/bilinear_form.lean

```diff
@@ -0,0 +1,146 @@
+/-
+Copyright (c) 2018 Andreas Swerdlow. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Author: Andreas Swerdlow
+-/
+
+import algebra.module
+
+universes u v
+
+class bilin_form (α : Type u) (β : Type v) [ring α] [add_comm_group β] [module α β] :=
```

Comment:
I think it would be nice if you also provide glue with the bilinear forms that are used in `linear_algebra/tensor_product`.

Reply:
```suggestion
variables {χ : Type*} [comm_ring χ] [module χ β] [bilin_form χ β]

instance base_ring_module : module χ χ := ring.to_module 
instance is_mod : module χ (β →ₗ[χ] χ) := linear_map.module

def to_linear_map : β →ₗ[χ] β →ₗ[χ] χ := 
linear_map.mk₂ χ (bilin χ) (bilin_add_left χ) bilin_smul_left (bilin_add_right χ) bilin_smul_right  
```

Would adding something like this be enough? I can't see a way of making the two concepts more linked as one requires the base ring to be commutative

Reply:
`variables {χ : Type*} [comm_ring χ] [module χ β] [bilin_form χ β]

instance base_ring_module : module χ χ := ring.to_module 
instance is_mod : module χ (β →ₗ[χ] χ) := linear_map.module

def to_linear_map : β →ₗ[χ] β →ₗ[χ] χ := 
linear_map.mk₂ χ (bilin χ) (bilin_add_left χ) bilin_smul_left (bilin_add_right χ) bilin_smul_right  
`

Would adding something like this be enough? I can't see a way to link the two concepts more as one requires the base ring to be commutative.

Reply:
Yes, that is one direction. But I think you should give a `linear_equiv`.

Reply:
Ok done the linear_equiv. Changed bilin_form from a class to a structure - should I do the same for sesq_form?